# HG changeset patch
# User Andreas Obergschwandtner <andreas.obergschwandtner@skidata.com>
# Date 1537519446 -7200
#      Fri Sep 21 10:44:06 2018 +0200
# Node ID 213da2c3ee54a558abd30b0230afa40d4b4e7f83
# Parent  80ee2541504ec08d65da9d968a18a00a69858ce0
Handle format error for GCC >= 7

diff --git a/hotspot/src/share/vm/adlc/output_c.cpp b/hotspot/src/share/vm/adlc/output_c.cpp
--- a/hotspot/src/share/vm/adlc/output_c.cpp
+++ b/hotspot/src/share/vm/adlc/output_c.cpp
@@ -436,9 +436,11 @@
   for (i = maxcycleused; i > 0; i /= 10)
     cycledigit++;
 
-  int maskdigit = 0;
-  for (i = rescount; i > 0; i /= 10)
+  int maskdigit = 1;
+  for (i = rescount / 10; i > 0; i /= 10)
     maskdigit++;
+  if (maskdigit > 10)
+    maskdigit = 10;
 
   static const char* pipeline_use_cycle_mask = "Pipeline_Use_Cycle_Mask";
   static const char* pipeline_use_element    = "Pipeline_Use_Element";
